### PR TITLE
Journal cleaner expanded

### PR DESCRIPTION
**Pull request by darthbeep**
Added functionality to delete rollable tables and folders in the journal cleaner.

### CODE REVIEW

File: dist/betteR20-5etools.user.js [hunk 1]
```diff
@@ -2786,7 +2788,31 @@ function baseTool () {
 						else return getAllJournalItems().filter(it => it);
 					}
 
+					// Allow for deleting tables as well
+					function getRollableTables() {
+						const tItems = [];
+						if ($cbTables.prop("checked")) {
```
Comment (TheGiddyLimit) [inline dist/betteR20-5etools.user.js]: Minor styling thing: flip this to be
```js
{
  if (!$cbTables.prop("checked")) return [];
  
  // ... rest of function body ... 
}
```
Means one less level of nesting, and one less thing to mentally track when reading the function. "Return early"!

File: dist/betteR20-5etools.user.js [hunk 2]
```diff
@@ -2786,7 +2788,31 @@ function baseTool () {
...
+							// Get a tableobject from the d20 thing and loop through it
+							tableObject = d20.Campaign.rollabletables;
```
Comment (TheGiddyLimit) [inline dist/betteR20-5etools.user.js]: Missing `const`

File: dist/betteR20-5etools.user.js [hunk 3]
```diff
@@ -2786,7 +2788,31 @@ function baseTool () {
...
+							for (i = 0; i < tableObject.length; i++) {
+								// If it looks confusing, it is. Just trust that I got the objects properly
+								tAttr = tableObject.at(i).attributes;
```
Comment (TheGiddyLimit) [inline dist/betteR20-5etools.user.js]: Missing `const`

File: dist/betteR20-5etools.user.js [hunk 4]
```diff
@@ -2786,7 +2788,31 @@ function baseTool () {
...
+								tObj = {
```
Comment (TheGiddyLimit) [inline dist/betteR20-5etools.user.js]: Missing `const`

File: dist/betteR20-5etools.user.js [hunk 5]
```diff
@@ -2786,7 +2788,31 @@ function baseTool () {
 						else return getAllJournalItems().filter(it => it);
 					}
 
+					// Allow for deleting tables as well
+					function getRollableTables() {
+						const tItems = [];
+						if ($cbTables.prop("checked")) {
+							// Get a tableobject from the d20 thing and loop through it
+							tableObject = d20.Campaign.rollabletables;
+							for (i = 0; i < tableObject.length; i++) {
```
Comment (TheGiddyLimit) [inline dist/betteR20-5etools.user.js]: Prefer `tableObject.map` over 90s for-i loops

File: dist/betteR20-5etools.user.js [hunk 6]
```diff
@@ -2797,6 +2823,7 @@ function baseTool () {
 								<input type="checkbox">
 								<span class="name readable">${it.path ? `${it.path} / ` : ""}${it.name}</span>
 								${it.archived ? "<span class=\"name readable\">(archived)</span>" : ""}
+								${it.table ? "<span class=\"name readable\">(table)</span>" : ""}
```
Comment (TheGiddyLimit) [inline dist/betteR20-5etools.user.js]: prefer template strings over escaping quotes: `<span class="name readable">(table)</span>`
(and the "archived" one too, please)

File: dist/betteR20-5etools.user.js (hunk 2 above)
Comment (TheGiddyLimit) [reply dist/betteR20-5etools.user.js]: Unsure why you're making this into a variable, anyway: use the full `d20.Campaign.rollabletables` everywhere instead. Don't rename things unless you have to--typing 10 extra characters is significantly easier than remembering function-specific aliases

File: dist/betteR20-core.user.js [hunk 7]
```diff
@@ -2027,6 +2027,7 @@ function baseTool () {
 				<div id="d20plus-quickdelete" title="BetteR20 - Journal Root Cleaner">
```
Comment (TheGiddyLimit) [inline dist/betteR20-core.user.js]: Didn't read the filename above, apologies--please _do not_ commit the `dist/` versions of the scripts; all this does is create unnecessary churn. 
All the comments above still apply, though

File: js/base-journal.js [hunk 8]
```diff
@@ -257,6 +258,7 @@ function d20plusJournal () {
 				return it.n && it.n === p;
 			});
 			if (!existing) return false;
+			if (!doDelete) return true;
```
Comment (TheGiddyLimit) [inline js/base-journal.js:261]: very nice!

File: js/base-tool.js [hunk 9]
```diff
@@ -15,7 +15,10 @@ function baseTool () {
 			html: `
 				<div id="d20plus-quickdelete" title="BetteR20 - Journal Root Cleaner">
 				<p>A list of characters and handouts in the journal folder root, which allows them to be quickly deleted.</p>
-				<label style="font-weight: bold">Root Only <input type="checkbox" class="cb-deep" checked></label>
+				<label style="font-weight: bold">Root Only <input type="radio" name="cb-mode" class="cb-shallow cb-mode"></label>
```
Comment (TheGiddyLimit) [inline js/base-tool.js]: prefer `class="bold"` over `style="..."`

**Review by TheGiddyLimit — CHANGES_REQUESTED** [review]

File: dist/betteR20-core.user.js (hunk 7 above)
Comment (darthbeep) [reply dist/betteR20-core.user.js]: Apologies, started this feature a long time ago before I figured out how to deal with dist and wasn't sure how to remove it.

**Comment by darthbeep** [comment]
I started this a while back before I knew how to deal with the dist files, and also when I didn't know much for the local/global stuff. While the code has been updated, I dealt with the dist files by ignoring them, but I should have deleted them. The code probably still has problems, but they're likely different problems from what you caught.

File: js/base-tool.js [hunk 10]
```diff
@@ -75,20 +117,60 @@ function baseTool () {
...
+
+						// If both are empty, check if item is a folder and return a folder type
+						if (d20plus.journal.checkDirExistsByPath(path.split(" / "))) return {type: "folder", id: itId, name: "", path:path, archived: false, folder: true}
 					}
 
 					function getJournalItems () {
-						if ($cbDeep.prop("checked")) return getRootJournalItems().filter(it => it);
-						else return getAllJournalItems().filter(it => it);
+						// For the root only option
+						if ($cbShallow.prop("checked")) return getRootJournalItems().filter(it => it);
```
Comment (TheGiddyLimit) [inline js/base-tool.js]: `filter(it => it)` can be `filter(Boolean)`, for a probably-nonexistent performance improvement 😎

File: js/base-tool.js [hunk 11]
```diff
@@ -98,6 +180,8 @@ function baseTool () {
 								<input type="checkbox">
 								<span class="name readable">${it.path ? `${it.path} / ` : ""}${it.name}</span>
 								${it.archived ? "<span class=\"name readable\">(archived)</span>" : ""}
+								${it.table ? "<span class=\"name readable\">(table)</span>" : ""}
```
Comment (TheGiddyLimit) [inline js/base-tool.js]: prefer template strings 
```
`<span class="name readable">(table)</span>`
```
over escaping quotes

**Review by TheGiddyLimit — APPROVED** [review]

**Pull request merged**